engine/src/main/java/org/kigalisim/engine/support/RecalcOperationBuilder.java

```diff
@@ -0,0 +1,274 @@
+        scopeEffective.orElse(null),
+        subtractRecharge.orElse(null)
+    );
+    strategies.add(newStrategy);
+    return this;
+  }
+
+  /**
+   * Propagate changes to consumption calculation.
+   *
+   * @return This builder for method chaining
+   */
+  public RecalcOperationBuilder thenPropagateToConsumption() {
+    if (!hasInitialRecalc) {
+      throw new IllegalStateException(
+          "Must call a recalc method before using thenPropagate methods");
+    }
+    strategies.add(new ConsumptionRecalcStrategy(scopeEffective.orElse(null)));
+    return this;
+  }
+
+  /**
+   * Propagate changes to sales calculation.
+   *
+   * @return This builder for method chaining
+   */
+  public RecalcOperationBuilder thenPropagateToSales() {
+    if (!hasInitialRecalc) {
+      throw new IllegalStateException(
+          "Must call a recalc method before using thenPropagate methods");
+    }
+    strategies.add(new SalesRecalcStrategy(scopeEffective.orElse(null)));
+    return this;
+  }
+
+  /**
+   * Propagate changes to recharge emissions calculation.
+   *
+   * @return This builder for method chaining
+   */
+  public RecalcOperationBuilder thenPropagateToRechargeEmissions() {
+    if (!hasInitialRecalc) {
+      throw new IllegalStateException(
+          "Must call a recalc method before using thenPropagate methods");
+    }
+    strategies.add(new RechargeEmissionsRecalcStrategy(scopeEffective.orElse(null)));
+    return this;
+  }
+
+  /**
+   * Propagate changes to end-of-life emissions calculation.
+   *
+   * @return This builder for method chaining
+   */
+  public RecalcOperationBuilder thenPropagateToEolEmissions() {
+    if (!hasInitialRecalc) {
+      throw new IllegalStateException(
+          "Must call a recalc method before using thenPropagate methods");
+    }
+    strategies.add(new EolEmissionsRecalcStrategy(scopeEffective.orElse(null)));
+    return this;
+  }
+
+  /**
+   * Propagate changes to retirement calculation.
+   *
+   * @return This builder for method chaining
+   */
+  public RecalcOperationBuilder thenPropagateToRetire() {
+    if (!hasInitialRecalc) {
+      throw new IllegalStateException(
+          "Must call a recalc method before using thenPropagate methods");
+    }
+    strategies.add(new RetireRecalcStrategy(scopeEffective.orElse(null)));
+    return this;
+  }
+
+  /**
+   * Build the RecalcOperation from the configured strategies.
+   *
+   * @return A RecalcOperation containing all configured strategies
+   * @throws IllegalStateException if no strategies have been configured
+   */
+  public RecalcOperation build() {
+    if (strategies.isEmpty()) {
+      throw new IllegalStateException("Must configure at least one recalculation strategy");
+    }
+    if (recalcKit.isPresent()) {
+      return new RecalcOperation(new ArrayList<>(strategies), recalcKit.get());
+    } else {
+      return new RecalcOperation(new ArrayList<>(strategies));
```

Comment:
Is this still needed? I would prefer just to have the version with RecalcKit, throwing an exception if recalcKit was not provided.